Makefile

```diff
@@ -41,24 +41,37 @@ INSTALL_FILES += $(shell cd sim && find waves -type f)
 
 $(foreach file, $(INSTALL_FILES), $(eval $(call declareInstallFile,$(file))))
 
+BRANCH ?= master
+
 .PHONY: checkout
+ifdef BENDER
+checkout: bender
+	./bender update
+	$(MAKE) bender-script
+else
 checkout:
-	git submodule update --init
 	./update-ips
+endif
 
 # generic clean and build targets for the platform
 .PHONY: clean
+
 ## Remove the RTL model files
 clean:
 	rm -rf $(VSIM_PATH)
-	cd sim && $(MAKE) clean
+	cd sim && $(MAKE) BENDER=$(BENDER) clean
 
 .PHONY: build
 ## Build the RTL model for vsim
 build:
+ifdef BENDER
+	@test -f Bender.lock || { echo "ERROR: Bender.lock file does not exist. Did you run make checkout in bender mode?"; exit 1; }
+	cd sim && $(MAKE) BENDER=bender all
+else
 	@[ "$$(ls -A ips/)" ] || { echo "ERROR: ips/ is an empty directory. Did you run ./update-ips?"; exit 1; }
-	cd sim && $(MAKE) lib build opt
+	cd sim && $(MAKE) all
```

Comment:
If you change this to `all`, then you should also remove `clean` from the `all` target it `sim/`. Anyway, it is a mistake from me to put `clean` into the `all` target.